### PR TITLE
cleanup some OOP code 

### PR DESCRIPTION
**Pull request by heejaechang**
### Customer scenario

There is no customer experience change

### Bugs this fixes

N/A

### Workarounds, if any

N/A

### Risk

N/A

### Performance impact

N/A

### Is this a regression from a previous update?

N/A

### Root cause analysis

just moving around code. no functional or perf changes.

### How was the bug found?

N/A

### CODE REVIEW

**Comment by heejaechang** [comment]
@jinujoseph @dotnet/roslyn-ide @dotnet/roslyn-analysis can i get code review? this doesn't change anything just moving code around.

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure I am not sure what is failing.

**Comment by heejaechang** [comment]
retest windows_debug_unit64_prtest  please

**Comment by heejaechang** [comment]
retest windows_release_unit64_prtest please

**Comment by heejaechang** [comment]
test failures known issue - https://github.com/dotnet/roslyn/issues/25931

I am working on it.

**Comment by heejaechang** [comment]
@Pilchie approval for 15.7?

**Comment by Pilchie** [comment]
I don't see the need to ship this in 15.7.  Is there more work planned that depends on it?  If not, I would suggest just putting this in master.

File: src/EditorFeatures/Test/SolutionCrawler/WorkCoordinatorTests.cs [hunk 1]
```diff
@@ -839,7 +839,7 @@ End Property
             }
         }
 
-        [Fact]
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/25931")]
```
Comment (sharwell) [inline src/EditorFeatures/Test/SolutionCrawler/WorkCoordinatorTests.cs]: ❗️ This is mitigated in #25938; no need to disable here.

File: src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.cs [hunk 2]
```diff
@@ -306,5 +277,33 @@ private static void SetNativeDllSearchDirectories()
                 }
             }
         }
+
+        public Task SynchronizePrimaryWorkspaceAsync(Checksum checksum, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.cs]: did these methods change?  or were these only moves?

Comment (heejaechang) [reply src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.cs]: moved

**Comment by heejaechang** [comment]
retest this please

**Comment by heejaechang** [comment]
retest this please

**Comment by heejaechang** [comment]
myget is not responding.

**Comment by heejaechang** [comment]
retest windows_release_unit64_prtest please

**Comment by heejaechang** [comment]
@jinujoseph this is master so 1 approval is all I need right?

**Comment by jinujoseph** [comment]
Yes

**Pull request merged**